uvloop/loop.pyx

```diff
@@ -1386,16 +1386,29 @@ cdef class Loop:
         """Create a Future object attached to the loop."""
         return self._new_future()
 
-    def create_task(self, coro):
+    def create_task(self, coro, *, name=None):
         """Schedule a coroutine object.
 
         Return a task object.
+
+        If name is not None, task.set_name(name) will be called if the task
+        object has the set_name attribute, true for default Task in Python 3.8.
         """
         self._check_closed()
         if self._task_factory is None:
             task = aio_Task(coro, loop=self)
         else:
             task = self._task_factory(self, coro)
+
+        # copied from asyncio.tasks._set_task_name (bpo-34270)
+        if name is not None:
+            try:
+                set_name = task.set_name
```

Comment:
Does this gain us something over the slightly more readable:

```
try:
    task.set_name(name)
except AttributeError:
    pass
```

Reply:
It won't mask an AttributeError coming from the `set_name` implementation. Which is unlikely but can still happen especially with custom task factories.

Reply:
Thanks for the suggestion and comments!

I would write something like this:

```python
if name is not None and hasattr(task, "set_name"):
    task.set_name(name)
```

But this is getting into details (`try..except` overhead? Two `O(1)` operations?) and stepping away from the copied code, so I think I'll merge as it is now.